src/test/kotlin/io/github/mojira/arisa/infrastructure/HelperMessagesTest.kt

```diff
@@ -0,0 +1,156 @@
+                "variable": [
+                    {
+                        "project": "mc",
+                        "value": "variable for MC",
+                        "localizedValues": {
+                            "ab": "abababab aba MC"
+                        }
+                    },
+                    {
+                        "project": "mcd",
+                        "value": "variable for MCD"
+                    }
+                ]
+            },
+            "messages": {
+                "normal": [
+                    {
+                        "project": "mc",
+                        "name": "Normal message",
+                        "message": "Normal message",
+                        "localizedMessages": {
+                            "ab": "Ababab abababa"
+                        },
+                        "fillname": []
+                    }
+                ],
+                "normal-list-filter": [
+                    {
+                        "project": ["mc", "mcd", "mcpe", "mcl", "mce"],
+                        "name": "Normal message with a list filter",
+                        "message": "Normal message with a list filter",
+                        "fillname": []
+                    }
+                ],
+                "with-variable": [
+                    {
+                        "project": ["mc", "mcd", "mcpe", "mcl", "mce"],
+                        "name": "With variable",
+                        "message": "With %variable%",
+                        "localizedMessages": {
+                            "ab": "Abab %variable%"
+                        },
+                        "fillname": []
+                    }
+                ],
+                "with-placeholder": [
+                    {
+                        "project": ["mc", "mcd", "mcpe", "mcl", "mce"],
+                        "name": "With placeholder",
+                        "message": "With %s%",
+                        "fillname": ["Placeholder"]
+                    }
+                ],
+                "i-am-a-bot": [
+                    {
+                        "project": ["mc", "mcd", "mcpe", "mcl", "mce"],
+                        "name": "I am a Bot",
+                        "message": "~{color:#888}-- I am a bot.{color}~",
+                        "localizedMessages": {
+                            "ab": "~{color:#888}-- A ba b aba.{color}~"
+                        },
+                        "fillname": []
+                    }
+                ]
+            }
+        }
+    """.trimIndent()
+    )!!
+    "should return Error when there is no such message" {
+        val result = messages.getSingleMessage("MC", "!@#%^&*")
+
+        result.shouldBeLeft()
+        result.a should { it is Error }
+        (result.a as Error).message shouldBe "Failed to find message for key !@#%^&* under project MC"
+    }
+    "should return Error when the message doesn't have a value for the project" {
```

Comment:
[Minor] Blank line between closing curly bracket and next test, to match style of other tests